tests/utilities/nidcpower_measurement/__init__.py

```diff
@@ -0,0 +1,50 @@
+"""NI-DCPower MeasurementLink test service."""
+import pathlib
+from typing import Iterable, Tuple
+
+import ni_measurementlink_service as nims
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "NIDCPowerMeasurement.serviceconfig",
+    version="0.1.0.0",
+    ui_file_paths=[
+        service_directory,
+    ],
+)
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration("pin_names", nims.DataType.PinArray1D, ["Pin1"])
+@measurement_service.configuration("multi_session", nims.DataType.Boolean, False)
+@measurement_service.output("session_names", nims.DataType.StringArray1D)
+@measurement_service.output("resource_names", nims.DataType.StringArray1D)
+@measurement_service.output("channel_lists", nims.DataType.StringArray1D)
+@measurement_service.output("channels_connected", nims.DataType.StringArray1D)
+def measure(
+    pin_names: Iterable[str],
+    multi_session: bool,
+) -> Tuple[Iterable[str], Iterable[str], Iterable[str], Iterable[str]]:
+    """NI-DCPower MeasurementLink test service."""
+    if multi_session:
+        with measurement_service.context.reserve_sessions(pin_names) as reservation:
+            with reservation.initialize_nidcpower_sessions() as session_infos:
+                connections = reservation.get_nidcpower_connections(pin_names)
+                assert [session_info.session is not None for session_info in session_infos]
+                return (
```

Comment:
I think these test services should call into the NI-DCPower session.

It would be best to perform an actual measurement, as suggested in the work item: "Consider using examples as a starting point, but simplifying (remove complex cancellation code, etc.)". 

This could look something like this for the multi-session case:
```python
for connection in connections:
    channel = connection.session.channels[connection.channel_name]
    channel.source_mode = nidcpower.SourceMode.SINGLE_POINT
    channel.output_function = nidcpower.OutputFunction.DC_VOLTAGE
    channel.current_limit = current_limit
    channel.voltage_level_range = 10.0
    channel.current_limit_range = 0.01
    channel.source_delay = hightime.timedelta(seconds=0.1)
    channel.voltage_level = 5.0

with ExitStack() as stack:
    for connection in connections:
        channel = connection.session.channels[connection.channel_name]
        stack.enter_context(channel.initiate())

    for connection in connections:
        channel = connection.session.channels[connection.channel_name]
        channel.wait_for_event(nidcpower.enums.Event.SOURCE_COMPLETE, timeout=10.0)

    for connection in connections:
        channel = connection.session.channels[connection.channel_name]
        measurement = channel.measure_multiple()[0]
        in_compliance = channel.query_in_compliance()
        measurement = measurement._replace(in_compliance=in_compliance)
```
Do not combine the for-loops. They are split up to parallelize operations across multiple channels.

You could return the measured values from the measure() call to allow the test to validate them, if you want. I would not recommend validating the simulated data in all test cases. Write one test case that validates the simulated data, and ignore it in the other test cases.

For the single-session version, consider extracting this into a helper function and passing in a list containing 1 connection.

Reply:
Okay, I initially thought that since the actual measurement will have driver calls and related logic, I need not add them necessarily to test our driver-specific session management APIs. But yeah, I accept the work item's description and I've added the actual measurement logic now.

Reply:
> Just to understand, if we have any particular reason to perform an actual measurement within the test measurement service?

The goal of this Python package is to enable users to write services that perform measurements, so our test services should perform measurements.